
From: William Lee Irwin III <wli@holomorphy.com>

Add atomic_sub_and_test() to sparc32, implemented in terms of
atomic_sub_return(), so reiser4 can be simultaneously microoptimized for
x86 and made to pass compilation testing on sparc32.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/asm-sparc/atomic.h |    1 +
 1 files changed, 1 insertion(+)

diff -puN include/asm-sparc/atomic.h~sparc32-add-atomic_sub_and_test include/asm-sparc/atomic.h
--- 25/include/asm-sparc/atomic.h~sparc32-add-atomic_sub_and_test	Tue Sep 14 17:18:30 2004
+++ 25-akpm/include/asm-sparc/atomic.h	Tue Sep 14 17:18:30 2004
@@ -46,6 +46,7 @@ extern void atomic_set(atomic_t *, int);
 #define atomic_inc_and_test(v) (atomic_inc_return(v) == 0)
 
 #define atomic_dec_and_test(v) (atomic_dec_return(v) == 0)
+#define atomic_sub_and_test(i, v) (atomic_sub_return(i, v) == 0)
 
 /* This is the old 24-bit implementation.  It's still used internally
  * by some sparc-specific code, notably the semaphore implementation.
_
